File: Client/Client/Data/Impl/AppointmentService.cs
﻿using System.Collections.Generic;
using System.Net;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using Client.Models;

namespace Client.Data.Impl
{
    public class AppointmentService : IAppointmentService
    {
        public const string _endpoint = "https://localhost:8083/SEP3";
        public async Task addAppointment(Appointment appointment)
        {
            using (HttpClient client = new HttpClient())
            {
                string serializedContent = JsonSerializer.Serialize(appointment);
                StringContent content = new StringContent(serializedContent, Encoding.UTF8, "application/json");

                await client.PostAsync($"{_endpoint}/appointment", content);
            }
        }

        public async Task<List<Appointment>> getAppointments(string serviceId)
        {
            using (HttpClient client = new HttpClient())
            {
                HttpResponseMessage response = await client.GetAsync($"{_endpoint}/appointments?serviceId={serviceId}");
                string reply = await response.Content.ReadAsStringAsync();
                
                List<Appointment> appointments =  JsonSerializer.Deserialize<List<Appointment>>(reply);
                return appointments;
            }
        }
        
        public async Task<List<Appointment>> getAppointmentsByUserId(string userId)
        {
            using (HttpClient client = new HttpClient())
            {
                HttpResponseMessage response = await client.GetAsync($"{_endpoint}/user-appointments?userId={userId}");
                string reply = await response.Content.ReadAsStringAsync();
                
                List<Appointment> appointments =  JsonSerializer.Deserialize<List<Appointment>>(reply);
                return appointments;
            }
        }

        public async Task deleteAppointment(string appointmentId)
        {
            using (HttpClient client = new HttpClient())
            { 
                await client.DeleteAsync($"{_endpoint}/appointment?appointmentId={appointmentId}");
            }
        }
    }
}

File: Server/Business/src/main/java/sep3/rest/appmngt/service/AppointmentService.java
package sep3.rest.appmngt.service;

import sep3.rest.appmngt.model.Appointment;

import java.io.IOException;
import java.util.List;

/**
 * The interface Appointment service.
 */
public interface AppointmentService  {
    /**
     * Add appointment appointment.
     *
     * @param appointment the appointment
     * @return the appointment
     * @throws IOException the io exception
     */
    Appointment addAppointment(Appointment appointment) throws IOException;

    /**
     * Gets appointments.
     *
     * @param serviceId the service id
     * @return the appointments
     * @throws IOException the io exception
     */
    List<Appointment> getAppointments(String serviceId) throws IOException;

    /**
     * Gets user appointments.
     *
     * @param userId the user id
     * @return the user appointments
     * @throws IOException the io exception
     */
    List<Appointment> getUserAppointments(String userId) throws IOException;

    /**
     * Delete appointment string.
     *
     * @param appointmentId the appointment id
     * @return the string
     * @throws IOException the io exception
     */
    String deleteAppointment(String appointmentId) throws IOException;
}
